src/dtemplate.d

```diff
@@ -1531,14 +1534,15 @@ public:
                     if (m == MATCHnomatch)
                     {
                         AggregateDeclaration ad = isAggregate(farg.type);
-                        if (ad && ad.aliasthis)
+                        if (ad && ad.aliasthis && argtype0 != att)
                         {
+                            if (!att && argtype0.checkAliasThisRec())   // Bugzilla 12537
+                                att = argtype0;
```

Comment:
This works with two mutually referring aliases, but what if there are three or more in the cycle?